packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -1049,6 +1064,35 @@ export function useGlobalStylesOutput() {
 			},
 		];
 
+		const processCSSNesting = ( css, blockSelector ) => {
+			let processedCSS = '';
+
+			// Split CSS nested rules.
+			const parts = css.split( '&' );
+			parts.forEach( ( part ) => {
+				processedCSS += ! part.includes( '{' )
+					? blockSelector + '{' + part + '}' // If the part doesn't contain braces, it applies to the root level.
+					: blockSelector + part; // Prepend the selector, which effectively replaces the "&" character.
+			} );
+			return processedCSS;
+		};
```

Comment:
Suggestion (non-blocking): This function can be moved outside the component — no need to recreate it on every re-render.

Reply:
Done, tested, and confirmed everything still works fine 👍